src/Main.elm

```diff
@@ -2,72 +2,137 @@ module Main exposing (main)
 
 import Browser
 import Data
+import UserGroup exposing (UserGroup)
+import ContactDetails
+import Settings
+import Tags
```

Comment:
I like the splitting this to separated modules - it makes the scope of each module clear and easier to read ❤️

Reply:
Yeah, I did the three tasks separately one after the other, so it made sense to split them into modules so I could focus on each task separately. But as I noted above, some of the form stuff like the view functions could obviously be DRYed up and generalised in a real-world project.